File: backend/app/schemas/audit.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.enums import AuditAction


class AuditLogEntry(BaseModel):
    id: uuid.UUID
    occurred_at: datetime
    actor_user_id: uuid.UUID | None
    actor_role: str | None
    action: AuditAction
    target_entity_type: str | None
    target_entity_id: uuid.UUID | None
    ip_address: str | None
    metadata_json: dict | None
    success: bool

    model_config = {"from_attributes": True}

File: frontend/src/types/audit.ts
// Mirrors app/models/enums.py AuditAction and schemas/audit.py AuditLogEntry.

export type AuditAction =
  | "login_success"
  | "login_failed"
  | "logout"
  | "create"
  | "read"
  | "update"
  | "delete"
  | "soft_delete"
  | "approve"
  | "reject"
  | "edit_ai_output"
  | "export_pdf"
  | "hitl_resolve";

export interface AuditLogEntry {
    
  id: string;
  occurred_at: string;
  actor_user_id: string | null;
  actor_role: string | null;
  action: AuditAction;
  target_entity_type: string | null;
  target_entity_id: string | null;
  ip_address: string | null;
  metadata_json: Record<string, unknown> | null;
  success: boolean;
}